serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/servicerecord/ServiceRecordFactory.java

```diff
@@ -49,147 +53,171 @@
+                                DocumentTypeService documentTypeService,
+                                SRRequestScope requestScope) {
         this.properties = properties;
         this.virksertService = virksertService;
         this.elmaLookupService = elmaLookupService;
         this.krrService = krrService;
         this.entityService = entityService;
         this.svarUtService = svarUtService;
+        this.processService = processService;
+        this.documentTypeService = documentTypeService;
+        this.requestScope = requestScope;
     }
 
-    @HystrixCommand(fallbackMethod = "createFiksErrorRecord")
-    public Optional<FiksWrapper> createFiksServiceRecord(String orgnr) {
-        Optional<Integer> adresseringNivaa = svarUtService.hasSvarUtAdressering(orgnr);
-        if (adresseringNivaa.isPresent()) {
-            String pem;
-            try {
-                pem = IOUtils.toString(properties.getSvarut().getCertificate().getInputStream(), StandardCharsets.UTF_8);
-            } catch (IOException e) {
-                log.error("Could not read certificate from {}", properties.getSvarut().getCertificate().toString());
-                throw new ServiceRegistryException(e);
-            }
-            FiksServiceRecord fiksServiceRecord = new FiksServiceRecord(orgnr, pem, properties.getSvarut().getServiceRecordUrl().toString());
-            return Optional.of(FiksWrapper.of(fiksServiceRecord, adresseringNivaa.get()));
+    public Optional<ServiceRecord> createArkivmeldingServiceRecord(String orgnr, String processIdentifier, Integer targetSecurityLevel) throws SecurityLevelNotFoundException, CertificateNotFoundException {
+        Optional<Process> optionalProcess = processService.findByIdentifier(processIdentifier);
+        if (optionalProcess.isPresent()) {
+            return Optional.of(createArkivmeldingServiceRecord(orgnr, targetSecurityLevel, optionalProcess.get()));
         }
         return Optional.empty();
     }
 
-    @SuppressWarnings("squid:S1172")
-    public Optional<FiksWrapper> createFiksErrorRecord(String orgnr, Throwable e) {
-        log.error("DPF service record failed", e);
-        return Optional.of(FiksWrapper.of(ErrorServiceRecord.create(DPF), 0));
-    }
-
-    @HystrixCommand(fallbackMethod = "createEduErrorRecord")
     @SuppressWarnings("squid:S1166")
-    public Optional<ServiceRecord> createEduServiceRecord(String orgnr) {
-        Endpoint ep;
-        try {
-            ep = elmaLookupService.lookup(NORWAY_PREFIX + orgnr);
-        } catch (EndpointUrlNotFound endpointUrlNotFound) {
-            log.debug(MarkerFactory.receiverMarker(orgnr),
-                    String.format("Attempted to lookup receiver in ELMA: %s", endpointUrlNotFound.getMessage()));
-            return Optional.empty();
+    public List<ServiceRecord> createArkivmeldingServiceRecords(String orgnr, Integer targetSecurityLevel) throws SecurityLevelNotFoundException, CertificateNotFoundException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: no.difi.meldingsutveksling.serviceregistry.exceptions.SecurityLevelNotFoundException, no.difi.meldingsutveksling.serviceregistry.CertificateNotFoundException [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1160)